Lib/test/test_re.py

```diff
@@ -1784,6 +1784,23 @@ def test_match_repr(self):
         )
         self.assertRegex(repr(second), pattern)
 
+        # Issue 39949: Verify truncation of long match
+        for matched_len in range(30,70):
+            string = ("abcde" * 20)[:matched_len]
+            match = re.match(r'\w*', string)
+
+            if len(string) <= 48:
```

Comment:
Can you make the test explicit? Right now it duplicates the logic in `_sre.c` so it's harder to see if something is wrong.

Reply:
I added 4 tests below which test the boundary of truncation for string / bytes. Let me know if I should do something else.